### DISCUSSION TITLE
Builds do not trigger automatically when condition inside my workflow is met

### DISCUSSION THREAD

**Discussion by Daeon97** [codemagic-ci-cd/codemagic-docs#1605] [Q&A] [answered]
I have this workflow

    workflows:
      run-tests-workflow:
        name: Run Tests Workflow
        max_build_duration: 70
        instance_type: mac_mini_m1
    
        environment:
          flutter: stable
    
        triggering:
          events:
            - pull_request
            - push
          branch_patterns:
            - pattern: "master"
              include: true
              source: false
            - pattern: "dev"
              include: true
              source: false
    
        scripts:
          - name: Get dependencies
            script: |
              flutter pub get
          - name: Run tests
            script: |
              flutter test
    
        publishing:
          email:
            recipients:
              - engelmmanuel@gmail.com
            notify:
              success: true
              failure: true
          slack:
            channel: '#publish'
            notify_on_build_start: true
            notify:
              success: true
              failure: true
          
But my run test workflow does not trigger on push or pull request to the specified branches. Am i doing anything wrong?

**Comment by dtrdic** [comment, marked as answer]
Hey Dion,
did you set up webhooks for automatic builds in your git repository?
I'm talking about this https://docs.codemagic.io/yaml-running-builds/webhooks/

**Reply by Daeon97** [reply]
Hello thanks for this. It solved my problem. Apparently i only needed to click on 'update webhook' in the webhooks tab of my project in codemagic. I was kinda confused because i didn't need to do this with my first codemagic project. The issue magically manifested itself on subsequent projects but simply clicking on 'update webhook' fixed it. Thanks alot @dtrdic

**Reply by dtrdic** [reply]
I'm glad to hear that:) cheers!
